File: src/test/java/com/ggardet/TestContainersConfiguration.java
package com.ggardet;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.springframework.context.annotation.Bean;
import org.testcontainers.containers.PostgreSQLContainer;

@TestConfiguration(proxyBeanMethods = false)
public class TestContainersConfiguration {
    @Bean
    @ServiceConnection
    public PostgreSQLContainer<?> postgreSQLContainer() {
        return (PostgreSQLContainer<?>) new PostgreSQLContainer("postgres:15.4-alpine3.18")
                .withDatabaseName("modularmonolith")
                .withUsername("modularmonolith")
                .withPassword("modularmonolith");
    }
}

File: src/test/kotlin/com/ggardet/TestContainersConfiguration.kt
package com.ggardet

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.testcontainers.containers.PostgreSQLContainer

@TestConfiguration(proxyBeanMethods = false)
class TestContainersConfiguration {
    
    @Bean
    @ServiceConnection
    fun postgreSQLContainer(): PostgreSQLContainer<*> {
        return PostgreSQLContainer("postgres:15.4-alpine3.18")
            .withDatabaseName("modularmonolith")
            .withUsername("modularmonolith")
            .withPassword("modularmonolith")
    }
}